app/activeproject.cpp

```diff
@@ -492,22 +488,26 @@ void ActiveProject::updateActiveLayer()
 {
   if ( !layerVisible( mActiveLayer.layer() ) )
   {
-    QgsMapLayer *defaultLayer = mRecordingLayerPM.layerFromLayerName( mAppSettings.defaultLayer() );
+    QgsMapLayer *defaultLayer = nullptr;
 
-    if ( !defaultLayer )
+    const QMap<QString, QgsMapLayer *> layers = mQgsProject->mapLayers();
+    for ( auto it = layers.cbegin(); it != layers.cend(); ++it )
     {
-      defaultLayer = mRecordingLayerPM.firstUsableLayer();
+      QgsMapLayer *layer = it.value();
+
+      // If it's a vector layer and visible, let's choose it
+      QgsVectorLayer *vectorLayer = qobject_cast<QgsVectorLayer *>( layer );
+      if ( vectorLayer && layerVisible( layer ) )
+      {
+        defaultLayer = layer;
+        break;
+      }
```

Comment:
The whole point of moving the functions to `InputUtils` was that we can use them here.. We need to check if the layer can be used for recording (`InputUtils::recordingAllowed`). Otherwise we end up in a situation like now:
<img width="333" alt="image" src="https://github.com/user-attachments/assets/1c35f924-4ae3-4288-a62a-44512a4fe17d" />